ksqldb-execution/src/main/java/io/confluent/ksql/execution/util/ExpressionTypeManager.java

```diff
@@ -153,14 +153,20 @@ public SqlType getExpressionSqlType(
     public Void visitArithmeticBinary(
         final ArithmeticBinaryExpression node,
         final Context context
-    ) {
+    ) throws KsqlException {
       process(node.getLeft(), context);
       final SqlType leftType = context.getSqlType();
 
       process(node.getRight(), context);
       final SqlType rightType = context.getSqlType();
 
-      final SqlType resultType = node.getOperator().resultType(leftType, rightType);
+      final SqlType resultType;
+      try {
+        resultType = node.getOperator().resultType(leftType, rightType);
+      } catch (KsqlException e) {
+        throw new KsqlException(String.format(
+                "Error processing expression: %s. %s", node.toString(), e.getMessage()));
```

Comment:
nice! thanks for the error message improvement :) might be good to also include `e` as the cause so that we don't lose the stack trace